zeebe/clients/java/src/main/java/io/camunda/zeebe/client/api/command/CommandWithCommunicationApiStep.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.api.command;
+
+import io.camunda.zeebe.client.api.ExperimentalApi;
+import java.util.Arrays;
+import java.util.List;
+
+public interface CommandWithCommunicationApiStep<T> {
+
+  String REST = "REST";
+  String GRPC = "GRPC";
+  List<String> AVAILABLE_COMMUNICATION_API = Arrays.asList(REST, GRPC);
```

Comment:
:thought_balloon: I initially used an `enum` with a String value to represent the possible communication API options. But I quickly realized that `enum`s are difficult to work with when using them for `String` config properties, as I needed to constantly get the String value of the enum for comparison.

Reply:
The benefit of an enum mostly is "type safety" and knowing the finite set of possible values, allowing for nicer switch statement support.

The communication API value is exposed to users via `ZeebeClientConfiguration#getDefaultCommunicationApi`. Do we need to expose it? If so, would we want to expose an enum here or stick with a String even if we'd use an enum internally?

Reply:
@tmetzke I think exposing an enum wouldn't bring much value, as configuration can be done through a `.properties` file or env vars, where a String will be used.

I'm fine with using an enum, but we would need to do some conversions whenever we want to perform some checks if a passed `String` property is correct. WDYT?